keras_cv/layers/preprocessing/grid_mask.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from tensorflow.keras import layers, backend
+
+
+class GridMask(layers.Layer):
+    """GridMask class for grid-mask augmentation. The expected images should be [0-255] pixel ranges.
+    
+    Args:
+        ratio: Grid mask ratio i.e if 0.5, grid and spacing will be equal.
+            Defaults to 0.5. It ranges from 0.0 to 1.0.
+        gridmask_size_ratio: Grid mask size, grid to image size ratio.
+            Defaults to 0.5. It ranges from 0.0 to 1.0.
+        rate: Float between 0 and 1. The probability of augmenting an input.
+            Defaults to 0.5.
+
+    Sample usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    random_gridmask = keras_cv.layers.preprocessing.GridMask(0.5, 0.5, rate=1.0)
+    augmented_images = random_gridmask(images)
+    ```
+    """
+    
+    def __init__(
+        self,
+        ratio=0.6,
+        gridmask_size_ratio=0.5,
+        rate=0.5,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.ratio = ratio
+        self.gridmask_size_ratio = gridmask_size_ratio
+        self.fill_value = 1 # TODO: make it adaptive, i.e. 'constant' or 'gaussian_noise'
```

Comment:
let's support fill value.  it can default to `constant`, with `fill_value=0`.

Reply:
I've just observed that in grid-mask, unlike cutout-random_erasing, the grid-block only appears to fill with black value only. So, a param like `fill_value` shouldn't necessarily be an option here for grid-mask augmentation. WDYT?

Reply:
I think that more generally many mask/mix fill ops could have an applicabile transform

Reply:
Eventually we plan to do some experimentation regarding mirror fill modes for all of these sorts of operations.  For now, this is fine.